packages/perseus/src/widgets/cs-program/cs-program-validator.test.ts

```diff
@@ -0,0 +1,54 @@
+import {csProgramValidator} from "./cs-program-validator";
+
+import type {PerseusCSProgramUserInput} from "../../validation.types";
+
+describe("csProgramValidator", () => {
+    it("is correct when the state from the iframe shows the status is correct", () => {
+        // Arrange
+        const state: PerseusCSProgramUserInput = {
+            status: "correct",
+            message: "Good job!",
+        };
+
+        // Act
+        const result = csProgramValidator(state);
+
+        // Assert
+        expect(result).toHaveBeenAnsweredCorrectly();
+    });
+
+    it("is incorrect when the state from the iframe shows the status is incorrect", () => {
+        // Arrange
+        const state: PerseusCSProgramUserInput = {
+            status: "incorrect",
+            message: "Try again!",
+        };
+
+        // Act
+        const result = csProgramValidator(state);
+
+        // Assert
+        expect(result).toEqual({
+            type: "points",
+            earned: 0,
+            total: 1,
+            message: "Try again!",
+        });
```

Comment:
If it's important that we check the message (and I'm not sure it is), I think we should update `toHaveBeenAnsweredIncorrectly` to optionally handle a message like `toHaveInvalidInput`. Otherwise I think we should switch this to `toHaveBeenAnsweredIncorrectly`.

Reply:
🤦‍♀️ Yep, totally meant to do that. Not sure how I skipped this. Thanks for catching it!